app/src/main/res/layout/avatar.xml

```diff
@@ -1,14 +1,21 @@
 <?xml version="1.0" encoding="utf-8"?>
-<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
-    xmlns:app="http://schemas.android.com/apk/res-auto"
-    android:layout_width="match_parent"
-    android:layout_height="match_parent"
-    android:orientation="vertical">
+<layout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto">
+    <data>
+        <variable
+            name="shape"
+            type="chat.rocket.android.helper.Constants"/>
+    </data>
+    <LinearLayout
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:orientation="vertical">
 
-    <com.facebook.drawee.view.SimpleDraweeView
-        android:id="@+id/image_avatar"
-        android:layout_width="40dp"
-        android:layout_height="40dp"
-        app:roundedCornerRadius="3dp" />
+        <com.facebook.drawee.view.SimpleDraweeView
+            android:id="@+id/image_avatar"
+            android:layout_width="40dp"
+            android:layout_height="40dp"
+            app:roundedCornerRadius='@{shape.AVATAR_SHAPE_CIRCLE ? @dimen/circle_avatar_corner_radius : @dimen/square_avatar_corner_radius}' />
```

Comment:
This is awesome!  Totally cool..... :-)

Reply:
Thanks :D
Learning how to do this thing and resolving subsequent errors took a lot of time but it paid off at the end :)